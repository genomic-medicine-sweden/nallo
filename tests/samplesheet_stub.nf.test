nextflow_pipeline {

    name "Test pipeline GENOMICMEDICINESWEDEN_NALLO"
    script "../main.nf"
    tag "PIPELINE"

    test("Test CADD with mock resources") {
        tag "samplesheet_stub"
        options "-stub"

        when {
            params {
                pipelines_testdata_base_path = 'https://raw.githubusercontent.com/genomic-medicine-sweden/test-datasets/9e0f40af8fb3ab7f8a3daecaeb03a0acda1e464e/'
                input                        = "$projectDir/assets/samplesheet.csv"
                outdir                       = "$outputDir"
                // Mock inputs for CADD
                cadd_resources               = "$projectDir/assets"
                cadd_prescored_indels        = "$projectDir/docs"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nallo_pipeline_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name
                ).match() }
            )
        }
    }

}
