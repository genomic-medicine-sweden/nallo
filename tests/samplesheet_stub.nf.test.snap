{
    "Test VEP with mock uncompressed cache": {
        "content": [
            161,
            {
                "ADD_FOUND_IN_TAG": {
                    "bcftools": 1.2,
                    "busybox_awk": "1.36.1"
                },
                "ADD_MOST_SEVERE_CSQ": {
                    "add_most_severe_consequence": 1.1,
                    "python": "3.8.3"
                },
                "ADD_MOST_SEVERE_PLI": {
                    "add_most_severe_pli": 1.1,
                    "python": "3.8.3"
                },
                "BCFTOOLS_CONCAT": {
                    "bcftools": 1.21
                },
                "BCFTOOLS_FILLTAGS": {
                    "bcftools": 1.2
                },
                "BCFTOOLS_INDEX": {
                    "bcftools": 1.21
                },
                "BCFTOOLS_MERGE": {
                    "bcftools": 1.21
                },
                "BCFTOOLS_NORM_MULTISAMPLE": {
                    "bcftools": 1.21
                },
                "BCFTOOLS_NORM_SINGLESAMPLE": {
                    "bcftools": 1.21
                },
                "BCFTOOLS_QUERY": {
                    "bcftools": 1.21
                },
                "BCFTOOLS_REHEADER": {
                    "bcftools": 1.21
                },
                "BCFTOOLS_SORT": {
                    "bcftools": 1.21
                },
                "BCFTOOLS_STATS": {
                    "bcftools": 1.21
                },
                "BCFTOOLS_VIEW": {
                    "bcftools": 1.21
                },
                "BEDTOOLS_MERGE": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_SORT": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_SPLIT": {
                    "bedtools": "2.31.1"
                },
                "CLEAN_SNIFFLES": {
                    "clean_sniffles": 1.0
                },
                "CRAMINO": {
                    "cramino": "0.14.5"
                },
                "CRAMINO_PHASED": {
                    "cramino": "0.14.5"
                },
                "CREATE_SAMPLES_FILE": {
                    "gawk": "5.1.0"
                },
                "CREATE_SAMPLES_HAPLOTYPES_FILE": {
                    "gawk": "5.1.0"
                },
                "DEEPVARIANT_RUNDEEPVARIANT": {
                    "deepvariant": "1.9.0"
                },
                "DEEPVARIANT_VCFSTATSREPORT": {
                    "deepvariant": "1.8.0"
                },
                "ECHTVAR_ANNO": {
                    "echtvar": "0.2.2"
                },
                "ENSEMBLVEP_FILTERVEP": {
                    "ensemblvep": 113.0
                },
                "ENSEMBLVEP_SNV": {
                    "ensemblvep": 110.0,
                    "tabix": 1.17
                },
                "ENSEMBLVEP_SV": {
                    "ensemblvep": 110.0,
                    "tabix": 1.17
                },
                "FASTQC": {
                    "fastqc": "0.12.1"
                },
                "GENMOD_ANNOTATE": {
                    "genmod": "3.10.1"
                },
                "GENMOD_COMPOUND": {
                    "genmod": "3.10.1"
                },
                "GENMOD_MODELS": {
                    "genmod": "3.10.1"
                },
                "GENMOD_SCORE": {
                    "genmod": "3.10.1"
                },
                "GFASTATS": {
                    "gfastats": "1.3.10"
                },
                "GLNEXUS": {
                    "glnexus": "1.4.3-0-gcecf42e"
                },
                "GUNZIP_FASTA": {
                    "gunzip": 1.13
                },
                "HIFIASM": {
                    "hifiasm": "0.25.0-r726"
                },
                "HIFICNV": {
                    "hificnv": "1.0.0-36e6461"
                },
                "LONGPHASE_HAPLOTAG": {
                    "longphase": "1.7.3"
                },
                "LONGPHASE_PHASE": {
                    "longphase": "1.7.3"
                },
                "MERGE_JSON": {
                    "merge_json_sample_files_into_family": 1.0,
                    "python": "3.8.3"
                },
                "MINIMAP2_ALIGN": {
                    "minimap2": "2.29-r1283"
                },
                "MINIMAP2_INDEX": {
                    "minimap2": "2.29-r1283"
                },
                "MODKIT_PILEUP": {
                    "modkit": "0.3.0"
                },
                "MOSDEPTH": {
                    "mosdepth": "0.3.10"
                },
                "PARAPHASE": {
                    "minimap2": "2.28-r1209",
                    "paraphase": "3.2.1",
                    "samtools": 1.21
                },
                "PEDDY": {
                    "peddy": "0.4.8"
                },
                "RELATE_RELATE": {
                    "somalier": "0.2.18"
                },
                "SAMPLESHEET_PED": {
                    "create_pedigree_file": 1.0,
                    "python": "3.8.3"
                },
                "SAMTOOLS_FAIDX": {
                    "samtools": 1.21
                },
                "SAMTOOLS_FASTQ": {
                    "samtools": 1.21
                },
                "SAMTOOLS_IMPORT": {
                    "samtools": 1.21
                },
                "SAMTOOLS_INDEX": {
                    "samtools": 1.21
                },
                "SAMTOOLS_INDEX_LONGPHASE": {
                    "samtools": 1.21
                },
                "SAMTOOLS_MERGE": {
                    "samtools": 1.21
                },
                "SAMTOOLS_SORT": {
                    "samtools": 1.21
                },
                "SAMTOOLS_VIEW": {
                    "samtools": 1.21
                },
                "SEVERUS": {
                    "severus": 1.5
                },
                "SNIFFLES": {
                    "sniffles": "1.0.12"
                },
                "SOMALIER_EXTRACT": {
                    "somalier": "0.2.18"
                },
                "SOMALIER_PED_FAMILY": {
                    "create_pedigree_file": 1.0,
                    "python": "3.8.3"
                },
                "SPLITUBAM": {
                    "splitubam": "0.1.1"
                },
                "STRANGER": {
                    "stranger": "0.9.5",
                    "tabix": 1.22
                },
                "SVDB_MERGE_BY_CALLER": {
                    "svdb": "2.8.2",
                    "bcftools": 1.21
                },
                "SVDB_MERGE_BY_FAMILY": {
                    "svdb": "2.8.2",
                    "bcftools": 1.21
                },
                "SVDB_QUERY": {
                    "svdb": "2.8.2"
                },
                "TABIX_BGZIPTABIX": {
                    "tabix": 1.21
                },
                "TABIX_ENSEMBLVEP_SNV": {
                    "tabix": 1.21
                },
                "TABIX_ENSEMBLVEP_SV": {
                    "tabix": 1.21
                },
                "TABIX_HIFICNV": {
                    "tabix": 1.21
                },
                "TABIX_LONGPHASE_PHASE": {
                    "tabix": 1.21
                },
                "TABIX_SEVERUS": {
                    "tabix": 1.21
                },
                "TAGBAM": {
                    "tagbam": "0.1.0"
                },
                "TRGT_GENOTYPE": {
                    "trgt": "3.0.0"
                },
                "TRGT_MERGE": {
                    "trgt": "3.0.0"
                },
                "WHATSHAP_STATS": {
                    "whatshap": 2.3,
                    "bgzip": 1.2,
                    "tabix": 1.2
                },
                "Workflow": {
                    "genomic-medicine-sweden/nallo": "v0.8.0dev"
                }
            },
            [
                "aligned_reads",
                "aligned_reads/HG002_Revio",
                "aligned_reads/HG002_Revio/HG002_Revio_haplotagged.bam",
                "aligned_reads/HG002_Revio/HG002_Revio_haplotagged.bam.bai",
                "assembly",
                "assembly/sample",
                "assembly/sample/HG002_Revio",
                "assembly/sample/HG002_Revio/HG002_Revio.bai",
                "assembly/sample/HG002_Revio/HG002_Revio.bam",
                "assembly/stats",
                "assembly/stats/HG002_Revio",
                "assembly/stats/HG002_Revio/HG002_Revio_haplotype_1.assembly_summary",
                "assembly/stats/HG002_Revio/HG002_Revio_haplotype_2.assembly_summary",
                "methylation",
                "methylation/modkit",
                "methylation/modkit/pileup",
                "methylation/modkit/pileup/HG002_Revio",
                "methylation/modkit/pileup/HG002_Revio/HG002_Revio_modkit_pileup.bed.gz",
                "methylation/modkit/pileup/HG002_Revio/HG002_Revio_modkit_pileup.bed.gz.tbi",
                "methylation/modkit/pileup/HG002_Revio/HG002_Revio_modkit_pileup.log",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_plots",
                "multiqc/multiqc_report.html",
                "paraphase",
                "paraphase/family",
                "paraphase/family/FAM",
                "paraphase/family/FAM/FAM_merged.json",
                "paraphase/family/FAM/FAM_paraphase_merged.vcf.gz",
                "paraphase/family/FAM/FAM_paraphase_merged.vcf.gz.tbi",
                "paraphase/sample",
                "paraphase/sample/HG002_Revio",
                "paraphase/sample/HG002_Revio/HG002_Revio.paraphase.bam",
                "paraphase/sample/HG002_Revio/HG002_Revio.paraphase.bam.bai",
                "paraphase/sample/HG002_Revio/HG002_Revio.paraphase.json",
                "paraphase/sample/HG002_Revio/HG002_Revio_paraphase_vcfs",
                "paraphase/sample/HG002_Revio/HG002_Revio_paraphase_vcfs/HG002_Revio_stub.vcf.gz",
                "paraphase/sample/HG002_Revio/HG002_Revio_paraphase_vcfs/HG002_Revio_stub.vcf.gz.tbi",
                "pedigree",
                "pedigree/family",
                "pedigree/family/FAM.ped",
                "phased_variants",
                "phased_variants/HG002_Revio",
                "phased_variants/HG002_Revio/HG002_Revio_phased.vcf.gz",
                "phased_variants/HG002_Revio/HG002_Revio_phased.vcf.gz.csi",
                "phased_variants/HG002_Revio/HG002_Revio_phased.vcf.gz.tbi",
                "pipeline_info",
                "pipeline_info/nallo_pipeline_software_mqc_versions.yml",
                "qc",
                "qc/bcftools_stats",
                "qc/bcftools_stats/HG002_Revio",
                "qc/bcftools_stats/HG002_Revio/HG002_Revio.bcftools_stats.txt",
                "qc/cramino",
                "qc/cramino/phased",
                "qc/cramino/phased/HG002_Revio",
                "qc/cramino/phased/HG002_Revio/HG002_Revio_cramino_aligned_phased.arrow",
                "qc/cramino/phased/HG002_Revio/HG002_Revio_cramino_aligned_phased.cramino.txt",
                "qc/cramino/unphased",
                "qc/cramino/unphased/HG002_Revio",
                "qc/cramino/unphased/HG002_Revio/HG002_Revio_cramino_aligned.arrow",
                "qc/cramino/unphased/HG002_Revio/HG002_Revio_cramino_aligned.cramino.txt",
                "qc/deepvariant_vcfstatsreport",
                "qc/deepvariant_vcfstatsreport/HG002_Revio",
                "qc/deepvariant_vcfstatsreport/HG002_Revio/HG002_Revio.visual_report.html",
                "qc/fastqc",
                "qc/fastqc/HG002_Revio",
                "qc/fastqc/HG002_Revio/HG002_Revio.html",
                "qc/fastqc/HG002_Revio/HG002_Revio.zip",
                "qc/mosdepth",
                "qc/mosdepth/HG002_Revio",
                "qc/mosdepth/HG002_Revio/HG002_Revio.global.dist.txt",
                "qc/mosdepth/HG002_Revio/HG002_Revio.per-base.bed.gz",
                "qc/mosdepth/HG002_Revio/HG002_Revio.per-base.bed.gz.csi",
                "qc/mosdepth/HG002_Revio/HG002_Revio.per-base.d4",
                "qc/mosdepth/HG002_Revio/HG002_Revio.quantized.bed.gz",
                "qc/mosdepth/HG002_Revio/HG002_Revio.quantized.bed.gz.csi",
                "qc/mosdepth/HG002_Revio/HG002_Revio.region.dist.txt",
                "qc/mosdepth/HG002_Revio/HG002_Revio.regions.bed.gz",
                "qc/mosdepth/HG002_Revio/HG002_Revio.regions.bed.gz.csi",
                "qc/mosdepth/HG002_Revio/HG002_Revio.summary.txt",
                "qc/mosdepth/HG002_Revio/HG002_Revio.thresholds.bed.gz",
                "qc/mosdepth/HG002_Revio/HG002_Revio.thresholds.bed.gz.csi",
                "qc/peddy",
                "qc/peddy/FAM",
                "qc/peddy/FAM/FAM.het_check.csv",
                "qc/peddy/FAM/FAM.het_check.png",
                "qc/peddy/FAM/FAM.html",
                "qc/peddy/FAM/FAM.ped_check.csv",
                "qc/peddy/FAM/FAM.ped_check.png",
                "qc/peddy/FAM/FAM.ped_check.rel-difference.csv",
                "qc/peddy/FAM/FAM.peddy.ped",
                "qc/peddy/FAM/FAM.sex_check.csv",
                "qc/peddy/FAM/FAM.sex_check.png",
                "qc/peddy/FAM/FAM.vs.html",
                "qc/phasing_stats",
                "qc/phasing_stats/HG002_Revio",
                "qc/phasing_stats/HG002_Revio/HG002_Revio_stats.blocks.gtf.gz",
                "qc/phasing_stats/HG002_Revio/HG002_Revio_stats.blocks.gtf.gz.tbi",
                "qc/phasing_stats/HG002_Revio/HG002_Revio_stats.stats.tsv",
                "qc/somalier",
                "qc/somalier/relate",
                "qc/somalier/relate/test",
                "qc/somalier/relate/test/test.html",
                "qc/somalier/relate/test/test.pairs.tsv",
                "qc/somalier/relate/test/test.samples.tsv",
                "repeats",
                "repeats/family",
                "repeats/family/FAM",
                "repeats/family/FAM/FAM_repeats_annotated.vcf.gz",
                "repeats/family/FAM/FAM_repeats_annotated.vcf.gz.tbi",
                "repeats/sample",
                "repeats/sample/HG002_Revio",
                "repeats/sample/HG002_Revio/HG002_Revio_spanning_trgt.bam",
                "repeats/sample/HG002_Revio/HG002_Revio_spanning_trgt.bam.bai",
                "repeats/sample/HG002_Revio/HG002_Revio_spanning_trgt.bam.csi",
                "repeats/sample/HG002_Revio/HG002_Revio_trgt.vcf.gz",
                "repeats/sample/HG002_Revio/HG002_Revio_trgt.vcf.gz.tbi",
                "snvs",
                "snvs/family",
                "snvs/family/FAM",
                "snvs/family/FAM/FAM_snvs_annotated_ranked_clinical.vcf.gz",
                "snvs/family/FAM/FAM_snvs_annotated_ranked_clinical.vcf.gz.tbi",
                "snvs/family/FAM/FAM_snvs_annotated_ranked_research.vcf.gz",
                "snvs/family/FAM/FAM_snvs_annotated_ranked_research.vcf.gz.tbi",
                "snvs/sample",
                "snvs/sample/HG002_Revio",
                "snvs/sample/HG002_Revio/HG002_Revio_deepvariant_snvs.vcf.gz",
                "snvs/sample/HG002_Revio/HG002_Revio_deepvariant_snvs.vcf.gz.tbi",
                "svs",
                "svs/family",
                "svs/family/FAM",
                "svs/family/FAM/FAM_svs_annotated_ranked_clinical.vcf.gz",
                "svs/family/FAM/FAM_svs_annotated_ranked_clinical.vcf.gz.tbi",
                "svs/family/FAM/FAM_svs_annotated_ranked_research.vcf.gz",
                "svs/family/FAM/FAM_svs_annotated_ranked_research.vcf.gz.tbi",
                "visualization_tracks",
                "visualization_tracks/HG002_Revio",
                "visualization_tracks/HG002_Revio/HG002_Revio_hificnv.bedgraph",
                "visualization_tracks/HG002_Revio/HG002_Revio_hificnv.depth.bw",
                "visualization_tracks/HG002_Revio/HG002_Revio_hificnv.maf.bw"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-07-28T17:18:38.7558902"
    },
    "Test CADD with mock resources": {
        "content": [
            174,
            {
                "ADD_FOUND_IN_TAG": {
                    "bcftools": 1.2,
                    "busybox_awk": "1.36.1"
                },
                "ADD_MOST_SEVERE_CSQ": {
                    "add_most_severe_consequence": 1.1,
                    "python": "3.8.3"
                },
                "ADD_MOST_SEVERE_PLI": {
                    "add_most_severe_pli": 1.1,
                    "python": "3.8.3"
                },
                "ANNOTATE_INDELS": {
                    "bcftools": 1.21
                },
                "BCFTOOLS_CONCAT": {
                    "bcftools": 1.21
                },
                "BCFTOOLS_FILLTAGS": {
                    "bcftools": 1.2
                },
                "BCFTOOLS_INDEX": {
                    "bcftools": 1.21
                },
                "BCFTOOLS_MERGE": {
                    "bcftools": 1.21
                },
                "BCFTOOLS_NORM_MULTISAMPLE": {
                    "bcftools": 1.21
                },
                "BCFTOOLS_NORM_SINGLESAMPLE": {
                    "bcftools": 1.21
                },
                "BCFTOOLS_QUERY": {
                    "bcftools": 1.21
                },
                "BCFTOOLS_REHEADER": {
                    "bcftools": 1.21
                },
                "BCFTOOLS_SORT": {
                    "bcftools": 1.21
                },
                "BCFTOOLS_STATS": {
                    "bcftools": 1.21
                },
                "BCFTOOLS_VIEW": {
                    "bcftools": 1.21
                },
                "BEDTOOLS_MERGE": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_SORT": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_SPLIT": {
                    "bedtools": "2.31.1"
                },
                "CADD": {
                    "cadd": "1.6.post1"
                },
                "CADD_TO_REFERENCE_CHRNAMES": {
                    "gawk": "5.3.0"
                },
                "CLEAN_SNIFFLES": {
                    "clean_sniffles": 1.0
                },
                "CRAMINO": {
                    "cramino": "0.14.5"
                },
                "CRAMINO_PHASED": {
                    "cramino": "0.14.5"
                },
                "CREATE_SAMPLES_FILE": {
                    "gawk": "5.1.0"
                },
                "CREATE_SAMPLES_HAPLOTYPES_FILE": {
                    "gawk": "5.1.0"
                },
                "DEEPVARIANT_RUNDEEPVARIANT": {
                    "deepvariant": "1.9.0"
                },
                "DEEPVARIANT_VCFSTATSREPORT": {
                    "deepvariant": "1.8.0"
                },
                "ECHTVAR_ANNO": {
                    "echtvar": "0.2.2"
                },
                "ENSEMBLVEP_FILTERVEP": {
                    "ensemblvep": 113.0
                },
                "ENSEMBLVEP_SNV": {
                    "ensemblvep": 110.0,
                    "tabix": 1.17
                },
                "ENSEMBLVEP_SV": {
                    "ensemblvep": 110.0,
                    "tabix": 1.17
                },
                "FASTQC": {
                    "fastqc": "0.12.1"
                },
                "GENMOD_ANNOTATE": {
                    "genmod": "3.10.1"
                },
                "GENMOD_COMPOUND": {
                    "genmod": "3.10.1"
                },
                "GENMOD_MODELS": {
                    "genmod": "3.10.1"
                },
                "GENMOD_SCORE": {
                    "genmod": "3.10.1"
                },
                "GFASTATS": {
                    "gfastats": "1.3.10"
                },
                "GLNEXUS": {
                    "glnexus": "1.4.3-0-gcecf42e"
                },
                "GUNZIP_FASTA": {
                    "gunzip": 1.13
                },
                "HIFIASM": {
                    "hifiasm": "0.25.0-r726"
                },
                "HIFICNV": {
                    "hificnv": "1.0.0-36e6461"
                },
                "LONGPHASE_HAPLOTAG": {
                    "longphase": "1.7.3"
                },
                "LONGPHASE_PHASE": {
                    "longphase": "1.7.3"
                },
                "MERGE_JSON": {
                    "merge_json_sample_files_into_family": 1.0,
                    "python": "3.8.3"
                },
                "MINIMAP2_ALIGN": {
                    "minimap2": "2.29-r1283"
                },
                "MINIMAP2_INDEX": {
                    "minimap2": "2.29-r1283"
                },
                "MODKIT_PILEUP": {
                    "modkit": "0.3.0"
                },
                "MOSDEPTH": {
                    "mosdepth": "0.3.10"
                },
                "PARAPHASE": {
                    "minimap2": "2.28-r1209",
                    "paraphase": "3.2.1",
                    "samtools": 1.21
                },
                "PEDDY": {
                    "peddy": "0.4.8"
                },
                "REFERENCE_TO_CADD_CHRNAMES": {
                    "gawk": "5.3.0"
                },
                "RELATE_RELATE": {
                    "somalier": "0.2.18"
                },
                "RENAME_CHRNAMES": {
                    "bcftools": 1.21
                },
                "SAMPLESHEET_PED": {
                    "create_pedigree_file": 1.0,
                    "python": "3.8.3"
                },
                "SAMTOOLS_FAIDX": {
                    "samtools": 1.21
                },
                "SAMTOOLS_FASTQ": {
                    "samtools": 1.21
                },
                "SAMTOOLS_IMPORT": {
                    "samtools": 1.21
                },
                "SAMTOOLS_INDEX": {
                    "samtools": 1.21
                },
                "SAMTOOLS_INDEX_LONGPHASE": {
                    "samtools": 1.21
                },
                "SAMTOOLS_MERGE": {
                    "samtools": 1.21
                },
                "SAMTOOLS_SORT": {
                    "samtools": 1.21
                },
                "SAMTOOLS_VIEW": {
                    "samtools": 1.21
                },
                "SEVERUS": {
                    "severus": 1.5
                },
                "SNIFFLES": {
                    "sniffles": "1.0.12"
                },
                "SOMALIER_EXTRACT": {
                    "somalier": "0.2.18"
                },
                "SOMALIER_PED_FAMILY": {
                    "create_pedigree_file": 1.0,
                    "python": "3.8.3"
                },
                "SPLITUBAM": {
                    "splitubam": "0.1.1"
                },
                "STRANGER": {
                    "stranger": "0.9.5",
                    "tabix": 1.22
                },
                "SVDB_MERGE_BY_CALLER": {
                    "svdb": "2.8.2",
                    "bcftools": 1.21
                },
                "SVDB_MERGE_BY_FAMILY": {
                    "svdb": "2.8.2",
                    "bcftools": 1.21
                },
                "SVDB_QUERY": {
                    "svdb": "2.8.2"
                },
                "TABIX_BGZIPTABIX": {
                    "tabix": 1.21
                },
                "TABIX_CADD": {
                    "tabix": 1.21
                },
                "TABIX_ENSEMBLVEP_SNV": {
                    "tabix": 1.21
                },
                "TABIX_ENSEMBLVEP_SV": {
                    "tabix": 1.21
                },
                "TABIX_HIFICNV": {
                    "tabix": 1.21
                },
                "TABIX_LONGPHASE_PHASE": {
                    "tabix": 1.21
                },
                "TABIX_SEVERUS": {
                    "tabix": 1.21
                },
                "TAGBAM": {
                    "tagbam": "0.1.0"
                },
                "TRGT_GENOTYPE": {
                    "trgt": "3.0.0"
                },
                "TRGT_MERGE": {
                    "trgt": "3.0.0"
                },
                "UNTAR_VEP_CACHE": {
                    "untar": 1.34
                },
                "WHATSHAP_STATS": {
                    "whatshap": 2.3,
                    "bgzip": 1.2,
                    "tabix": 1.2
                },
                "Workflow": {
                    "genomic-medicine-sweden/nallo": "v0.8.0dev"
                }
            },
            [
                "aligned_reads",
                "aligned_reads/HG002_Revio",
                "aligned_reads/HG002_Revio/HG002_Revio_haplotagged.bam",
                "aligned_reads/HG002_Revio/HG002_Revio_haplotagged.bam.bai",
                "assembly",
                "assembly/sample",
                "assembly/sample/HG002_Revio",
                "assembly/sample/HG002_Revio/HG002_Revio.bai",
                "assembly/sample/HG002_Revio/HG002_Revio.bam",
                "assembly/stats",
                "assembly/stats/HG002_Revio",
                "assembly/stats/HG002_Revio/HG002_Revio_haplotype_1.assembly_summary",
                "assembly/stats/HG002_Revio/HG002_Revio_haplotype_2.assembly_summary",
                "methylation",
                "methylation/modkit",
                "methylation/modkit/pileup",
                "methylation/modkit/pileup/HG002_Revio",
                "methylation/modkit/pileup/HG002_Revio/HG002_Revio_modkit_pileup.bed.gz",
                "methylation/modkit/pileup/HG002_Revio/HG002_Revio_modkit_pileup.bed.gz.tbi",
                "methylation/modkit/pileup/HG002_Revio/HG002_Revio_modkit_pileup.log",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_plots",
                "multiqc/multiqc_report.html",
                "paraphase",
                "paraphase/family",
                "paraphase/family/FAM",
                "paraphase/family/FAM/FAM_merged.json",
                "paraphase/family/FAM/FAM_paraphase_merged.vcf.gz",
                "paraphase/family/FAM/FAM_paraphase_merged.vcf.gz.tbi",
                "paraphase/sample",
                "paraphase/sample/HG002_Revio",
                "paraphase/sample/HG002_Revio/HG002_Revio.paraphase.bam",
                "paraphase/sample/HG002_Revio/HG002_Revio.paraphase.bam.bai",
                "paraphase/sample/HG002_Revio/HG002_Revio.paraphase.json",
                "paraphase/sample/HG002_Revio/HG002_Revio_paraphase_vcfs",
                "paraphase/sample/HG002_Revio/HG002_Revio_paraphase_vcfs/HG002_Revio_stub.vcf.gz",
                "paraphase/sample/HG002_Revio/HG002_Revio_paraphase_vcfs/HG002_Revio_stub.vcf.gz.tbi",
                "pedigree",
                "pedigree/family",
                "pedigree/family/FAM.ped",
                "phased_variants",
                "phased_variants/HG002_Revio",
                "phased_variants/HG002_Revio/HG002_Revio_phased.vcf.gz",
                "phased_variants/HG002_Revio/HG002_Revio_phased.vcf.gz.csi",
                "phased_variants/HG002_Revio/HG002_Revio_phased.vcf.gz.tbi",
                "pipeline_info",
                "pipeline_info/nallo_pipeline_software_mqc_versions.yml",
                "qc",
                "qc/bcftools_stats",
                "qc/bcftools_stats/HG002_Revio",
                "qc/bcftools_stats/HG002_Revio/HG002_Revio.bcftools_stats.txt",
                "qc/cramino",
                "qc/cramino/phased",
                "qc/cramino/phased/HG002_Revio",
                "qc/cramino/phased/HG002_Revio/HG002_Revio_cramino_aligned_phased.arrow",
                "qc/cramino/phased/HG002_Revio/HG002_Revio_cramino_aligned_phased.cramino.txt",
                "qc/cramino/unphased",
                "qc/cramino/unphased/HG002_Revio",
                "qc/cramino/unphased/HG002_Revio/HG002_Revio_cramino_aligned.arrow",
                "qc/cramino/unphased/HG002_Revio/HG002_Revio_cramino_aligned.cramino.txt",
                "qc/deepvariant_vcfstatsreport",
                "qc/deepvariant_vcfstatsreport/HG002_Revio",
                "qc/deepvariant_vcfstatsreport/HG002_Revio/HG002_Revio.visual_report.html",
                "qc/fastqc",
                "qc/fastqc/HG002_Revio",
                "qc/fastqc/HG002_Revio/HG002_Revio.html",
                "qc/fastqc/HG002_Revio/HG002_Revio.zip",
                "qc/mosdepth",
                "qc/mosdepth/HG002_Revio",
                "qc/mosdepth/HG002_Revio/HG002_Revio.global.dist.txt",
                "qc/mosdepth/HG002_Revio/HG002_Revio.per-base.bed.gz",
                "qc/mosdepth/HG002_Revio/HG002_Revio.per-base.bed.gz.csi",
                "qc/mosdepth/HG002_Revio/HG002_Revio.per-base.d4",
                "qc/mosdepth/HG002_Revio/HG002_Revio.quantized.bed.gz",
                "qc/mosdepth/HG002_Revio/HG002_Revio.quantized.bed.gz.csi",
                "qc/mosdepth/HG002_Revio/HG002_Revio.region.dist.txt",
                "qc/mosdepth/HG002_Revio/HG002_Revio.regions.bed.gz",
                "qc/mosdepth/HG002_Revio/HG002_Revio.regions.bed.gz.csi",
                "qc/mosdepth/HG002_Revio/HG002_Revio.summary.txt",
                "qc/mosdepth/HG002_Revio/HG002_Revio.thresholds.bed.gz",
                "qc/mosdepth/HG002_Revio/HG002_Revio.thresholds.bed.gz.csi",
                "qc/peddy",
                "qc/peddy/FAM",
                "qc/peddy/FAM/FAM.het_check.csv",
                "qc/peddy/FAM/FAM.het_check.png",
                "qc/peddy/FAM/FAM.html",
                "qc/peddy/FAM/FAM.ped_check.csv",
                "qc/peddy/FAM/FAM.ped_check.png",
                "qc/peddy/FAM/FAM.ped_check.rel-difference.csv",
                "qc/peddy/FAM/FAM.peddy.ped",
                "qc/peddy/FAM/FAM.sex_check.csv",
                "qc/peddy/FAM/FAM.sex_check.png",
                "qc/peddy/FAM/FAM.vs.html",
                "qc/phasing_stats",
                "qc/phasing_stats/HG002_Revio",
                "qc/phasing_stats/HG002_Revio/HG002_Revio_stats.blocks.gtf.gz",
                "qc/phasing_stats/HG002_Revio/HG002_Revio_stats.blocks.gtf.gz.tbi",
                "qc/phasing_stats/HG002_Revio/HG002_Revio_stats.stats.tsv",
                "qc/somalier",
                "qc/somalier/relate",
                "qc/somalier/relate/test",
                "qc/somalier/relate/test/test.html",
                "qc/somalier/relate/test/test.pairs.tsv",
                "qc/somalier/relate/test/test.samples.tsv",
                "repeats",
                "repeats/family",
                "repeats/family/FAM",
                "repeats/family/FAM/FAM_repeats_annotated.vcf.gz",
                "repeats/family/FAM/FAM_repeats_annotated.vcf.gz.tbi",
                "repeats/sample",
                "repeats/sample/HG002_Revio",
                "repeats/sample/HG002_Revio/HG002_Revio_spanning_trgt.bam",
                "repeats/sample/HG002_Revio/HG002_Revio_spanning_trgt.bam.bai",
                "repeats/sample/HG002_Revio/HG002_Revio_spanning_trgt.bam.csi",
                "repeats/sample/HG002_Revio/HG002_Revio_trgt.vcf.gz",
                "repeats/sample/HG002_Revio/HG002_Revio_trgt.vcf.gz.tbi",
                "snvs",
                "snvs/family",
                "snvs/family/FAM",
                "snvs/family/FAM/FAM_snvs_annotated_ranked_clinical.vcf.gz",
                "snvs/family/FAM/FAM_snvs_annotated_ranked_clinical.vcf.gz.tbi",
                "snvs/family/FAM/FAM_snvs_annotated_ranked_research.vcf.gz",
                "snvs/family/FAM/FAM_snvs_annotated_ranked_research.vcf.gz.tbi",
                "snvs/sample",
                "snvs/sample/HG002_Revio",
                "snvs/sample/HG002_Revio/HG002_Revio_deepvariant_snvs.vcf.gz",
                "snvs/sample/HG002_Revio/HG002_Revio_deepvariant_snvs.vcf.gz.tbi",
                "svs",
                "svs/family",
                "svs/family/FAM",
                "svs/family/FAM/FAM_svs_annotated_ranked_clinical.vcf.gz",
                "svs/family/FAM/FAM_svs_annotated_ranked_clinical.vcf.gz.tbi",
                "svs/family/FAM/FAM_svs_annotated_ranked_research.vcf.gz",
                "svs/family/FAM/FAM_svs_annotated_ranked_research.vcf.gz.tbi",
                "visualization_tracks",
                "visualization_tracks/HG002_Revio",
                "visualization_tracks/HG002_Revio/HG002_Revio_hificnv.bedgraph",
                "visualization_tracks/HG002_Revio/HG002_Revio_hificnv.depth.bw",
                "visualization_tracks/HG002_Revio/HG002_Revio_hificnv.maf.bw"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-07-28T17:17:55.798651085"
    }
}